amxmodx/CLang.cpp

```diff
@@ -332,167 +480,60 @@ int CLangMngr::MergeDefinitionFile(const char *file)
 	/** Checks if there is an existing entry with same time stamp. */
 	time_t timeStamp;
 	if (FileList.retrieve(file, &timeStamp) && fileStat.st_mtime == timeStamp)
+	{
 		return -1;
+	}
 
 	/** If yes, it either means that the entry doesn't exist or the existing entry needs to be updated. */
 	FileList.replace(file, fileStat.st_mtime);
 
-	FILE* fp = fopen(file, "rt");
-	if (!fp)
+	Data.currentFile = file;
+
+	unsigned int line, col;
+	bool result = textparsers->ParseFile_INI(file, static_cast<ITextListener_INI*>(this), &line, &col);
+
+	if (!result)
 	{
 		AMXXLOG_Log("[AMXX] Failed to re-open dictionary file: %s", file);
 		return 0;
 	}
 
-	// Allocate enough memory to store everything
-	bool multiline = 0;
-	int pos = 0, line = 0;
-	CQueue<sKeyDef> Defq;
-	String buf;
-	char language[3];
-	sKeyDef tmpEntry = {NULL, 0};
-
-	while (!feof(fp))
-	{
-		line++;
-		buf._fread(fp);
-		buf.trim();
-		if (buf[0] == 0)
-			continue;
-		if ((buf[0] == ';') || (buf[0] == '/' && buf[1] == '/'))
-			continue;
-
-		/* Check for BOM markings, which is only relevant on the first line.
-		* Not worth it, but it could be moved out of the loop.
-		*/
-		if (line == 1 && (buf[0] == (char)0xEF && buf[1] == (char)0xBB && buf[2] == (char)0xBF))
-		{
-			buf.erase(0, 3);
-		}
-
-		if (buf[0] == '[' && buf.size() >= 3)
-		{
-			if (multiline)
-			{
-				AMXXLOG_Log("New section, multiline unterminated (file \"%s\" line %d)", file, line);
-				tmpEntry.key = -1;
-				tmpEntry.definition = NULL;
-			}
-			
-			if (!Defq.empty())
-			{
-				MergeDefinitions(language, Defq);
-			}
-			
-			language[0] = buf[1];
-			language[1] = buf[2];
-			language[2] = 0;
-		} else {
-			if (!multiline)
-			{
-				pos = buf.find('=');
-				
-				if (pos > String::npos)
-				{
-					String key;
-					key.assign(buf.substr(0, pos).c_str());
-					String def;
-					def.assign(buf.substr(pos + 1).c_str());
-					key.trim();
-					key.toLower();
-					int iKey = GetKeyEntry(key);
-					if (iKey == -1)
-						iKey = AddKeyEntry(key);
-					tmpEntry.key = iKey;
-					tmpEntry.definition = new String;
-					tmpEntry.definition->assign(def.c_str());
-					tmpEntry.definition->trim();
-					reparse_color(tmpEntry.definition);
-					tmpEntry.definition->reparse_newlines();
-					Defq.push(tmpEntry);
-					tmpEntry.key = -1;
-					tmpEntry.definition = NULL;
-				} else {
-					pos = buf.find(':');
-					
-					if (pos > String::npos)
-					{
-						String key;
-						key.assign(buf.substr(0, pos).c_str());;
-						key.trim();
-						key.toLower();
-						int iKey = GetKeyEntry(key);
-						if (iKey == -1)
-							iKey = AddKeyEntry(key);
-						tmpEntry.key = iKey;
-						tmpEntry.definition = new String;
-						multiline = true;
-					} else {
-						//user typed a line with no directives
-						AMXXLOG_Log("Invalid multi-lingual line (file \"%s\" line %d)", file, line);
-					}
-				}
-			} else {
-				if (buf[0] == ':')
```

Comment:
FYI, removing multiline support in dictionary files is an undocumented breaking change.